java/src/com/ibm/streamsx/topology/generator/spl/SPLGenerator.java

```diff
@@ -83,6 +98,406 @@ public String generateSPL(JsonObject graph) throws IOException {
+        
+        // Create the outputs of the invocation
+        JsonArray outputs = new JsonArray();
+        for(JsonObject endOp : startsEndsAndOperators.get(1)){
+            GraphUtilities.outputs(endOp, output -> outputs.add(output));
+        } 
+        compositeInvocation.add("outputs", outputs);
+        
+        return compositeInvocation;
+    }
+
+    private JsonObject createParallelCompositeInvocation(JsonObject opDefinition, List<List<JsonObject>> startsEndsAndOperators) {
+        JsonObject compositeInvocation = createCompositeInvocation(opDefinition, startsEndsAndOperators);
+        
+        // Create object with parallel information of input ports
+        JsonObject parallelInfo = new JsonObject();
+        JsonArray broadcastPorts = new JsonArray();
+        JsonArray partitionedPorts = new JsonArray();
+        
+        parallelInfo.add("broadcastPorts", broadcastPorts);
+        parallelInfo.add("partitionedPorts", partitionedPorts);
+        
+        for(JsonObject startOp : startsEndsAndOperators.get(0)){
+            if(startOp.has("config") && startOp.get("config").getAsJsonObject().has(OpProperties.WIDTH)){
+                JsonElement width = startOp.get("config").getAsJsonObject().get(OpProperties.WIDTH);
+                parallelInfo.add(OpProperties.WIDTH, width);
+            }
+            
+            // If it's a physical source start operator, ignore its output port, it has no partition information.
+            if(startOp.has("config") && (hasAny(object(startOp, "config"), compOperatorStarts)))
+                continue;
+            
+            // Otherwise, get the partition information.
+            JsonObject outputPort = array(startOp, "outputs").get(0).getAsJsonObject();
+            JsonObject inputPort = array(startOp, "inputs").get(0).getAsJsonObject();
+            
+            // Set the width if it was contained in the output port.
+            if(outputPort.has(OpProperties.WIDTH))
+                parallelInfo.add(OpProperties.WIDTH, outputPort.get(OpProperties.WIDTH));
+            
+            
+            if(jstring(outputPort, PortProperties.ROUTING).equals(Routing.BROADCAST.toString())){
+                broadcastPorts.add(inputPort.get("name"));
+            }
+            
+            if(outputPort.has(PortProperties.PARTITIONED) && jboolean(outputPort, PortProperties.PARTITIONED)){
+                JsonObject partitionInfo = new JsonObject();
+                partitionInfo.add("name", inputPort.get("name"));
+                partitionInfo.add(PortProperties.PARTITION_KEYS, outputPort.get(PortProperties.PARTITION_KEYS));
+                partitionedPorts.add(partitionInfo);
+                
+                // There is at least one partitioned port, therefore it is partitioned.
+                compositeInvocation.addProperty("partitioned", true);
+            }      
+                
+        } 
+        
+        compositeInvocation.add("parallelInfo", parallelInfo);
```

Comment:
Can `parallelOperator` ever be false if there is a `parallelInfo`. 

Maybe later cleanup, but is the presence of `parallelInfo` sufficient to indicate the op is parallel.